### PR TITLE
M69, Metal and more 🤘

### PR DESCRIPTION
**Pull request by saghul**
The M69 update (TM)

- Updated WebRTC to M69
- Metal renderer for iOS
- Promise APIs only
- W3C-ish permissions API

All feedback is welcome. I'll let this sit here for a bit before merging.

### CODE REVIEW

File: MediaDevices.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class MediaDevices extends EventTarget(MEDIA_DEVICES_EVENTS) {
+    // TODO: implement.
+    ondevicechange: ?Function;
+
+    /**
+     * W3C "Media Capture and Streams" compatible {@code enumerateDevices}
+     * implementation.
+     */
+    enumerateDevices() {
+        return new Promise(resolve => WebRTCModule.enumerateDevices(ressolve));
```
Comment (brunsy) [inline MediaDevices.js]: "ressolve" -> resolve

Comment (saghul) [reply MediaDevices.js]: Doh, nice catch!

**Comment by zxcpoiu** [comment]
@saghul :muscle: :muscle:

**Comment by brunsy** [comment]
iOS GUM is throwing a ```WebRTCModule.requestPermission is not a function```. 
Android is working great

**Comment by saghul** [comment]
@brunsy Interesting. I cannot reproduce that. Can you please clean your build and try again? The function is certainly there...

**Comment by saghul** [comment]
@brunsy ah I see now! You are not using CocoaPods, right? I forgot to add the permissions file to the Xcode project. There is a new commit in this PR which should solve your problem, can you please pull and test again? Thanks for the report!

**Comment by brunsy** [comment]
Yeah, we are not using CoccaPods. Thanks @saghul - that last commit did the trick, working great on iOS now.

**Comment by saghul** [comment]
@zxcpoiu This is now merged, but I want to look at some of the PRs and update the sample code in the README before the next release, so please don't make one yet.

**Pull request merged**